src/mono/mono/mini/CMakeLists.txt

```diff
@@ -381,6 +381,42 @@ if(NOT DISABLE_SHARED_LIBS)
     add_library(monosgen-shared-dac SHARED "mini-windows-dlldac.c")
     set_target_properties(monosgen-shared-dac PROPERTIES OUTPUT_NAME ${MONO_SHARED_LIB_NAME}-dac)
   endif()
+
+  if(BUILD_DARWIN_FRAMEWORKS)
+    if(TARGET_DARWIN)
+      string(REPLACE ";" "*" mono-components-objects-nowhitespace "${mono-components-objects}")
```

Comment:
Why is this in terms of the component objects, and not the `mono-component-${component}-static` and `mono-component-${component}-stub-static` .a libraries?

Also I don't understand what's going on with the whitespace business.  Maybe it needs a comment?